### PR TITLE
new icon: matplotlib (original, original-wordmark, plain, plain-wordmark)

### LINKED ISSUE #859: Icon request: Matplotlib
### About the icon

Matplotlib is an essential Python library used to creating plots.

### Links and sources

https://matplotlib.org/

### Additional information

Thank you

### PR DESCRIPTION
**Pull request by BenSouchet**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened
- [x] PR name matches the format *new icon: <i>Icon name</i> (<i>versions separated by comma</i>)*. More details [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#overview)
- [x] PR's base is the `develop` branch.
- [x] Your icons are inside a folder as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#organizational-guidelines)
- [x] SVG matches the standards laid out [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#svgStandards)
- [x] A new object is added in the `devicon.json` file as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#-updating-the-deviconjson-)

**This PR closes #859**

## Link to prove your SVG is correct and up-to-date.

https://github.com/matplotlib/matplotlib

### CODE REVIEW

**Comment by BenSouchet** [comment]
@Panquesito7 Since the `original` icons won't be used in icomoon can you skip the error ?

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Could you please fix the issues mentioned by the bot? Let us know if you need any help with that! Thank you for your contribution. 😃


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

**Comment by BenSouchet** [comment]
@Panquesito7 Like I said in my previous message:
> Since the `original` icons **won't be used in icomoon** can you skip / ignore the error ?
Because converting the dotted stroke will increase a lot the SVG files for no benefices.

**Comment by Panquesito7** [comment]
> @Panquesito7 Since the `original` icons won't be used in icomoon can you skip the error ? Because converting the dotted stroke will increase a lot the SVG files for no benefices.

Hey! Not sure if we can actually skip that error. I think we should still follow these rules.
@Snailedlt, what do you think we should do here?

**Comment by BenSouchet** [comment]
This test exists because **icomoon** can't handle strokes but these two `original` icons won't be used by **icomoon**.

The check (Bot) should ignore the `original` versions if `plain` versions exists.

**Comment by Snailedlt** [comment]
@Panquesito7 I think what @BenSouchet is saying makes sense, but I'm a bit afraid to voich for it since I don't know the intricacies.

We can try with peek:bot, if it still looks okey after that imo, we can merge. Then we'll just keep an eye out for that icon, and revert if it causes any issues :)

File: icons/matplotlib/matplotlib-plain-wordmark.svg
Comment (Snailedlt) [inline icons/matplotlib/matplotlib-plain-wordmark.svg:1]: Please merge the paths if possible in accordance to the [SVG Standards](https://github.com/devicons/devicon/wiki/SVG-Standards)

File: icons/matplotlib/matplotlib-plain.svg
Comment (Snailedlt) [inline icons/matplotlib/matplotlib-plain.svg:1]: Please merge the paths if possible in accordance to the [SVG Standards](https://github.com/devicons/devicon/wiki/SVG-Standards)

**Review by Snailedlt — CHANGES_REQUESTED** [review]
The paths for plain versions should be merged into one, other than that it looks good :)

**Comment by BenSouchet** [comment]
> The paths for plain versions should be merged into one, other than that it looks good :)

Paths optimized

File: icons/matplotlib/matplotlib-plain-wordmark.svg
Comment (Snailedlt) [inline icons/matplotlib/matplotlib-plain-wordmark.svg:1]: Still multiple paths here. You can try using [SVGOMG](https://jakearchibald.github.io/svgomg/) or this SVG optimizer -> https://www.ofoct.com/image-converter/svg-optimizer.html. 
Both websites have a `merge Paths` option

File: icons/matplotlib/matplotlib-plain.svg
Comment (Snailedlt) [inline icons/matplotlib/matplotlib-plain.svg:1]: This one also still has multiple paths

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Still multiple paths. Everything else looks good as far as I can tell

File: icons/matplotlib/matplotlib-plain-wordmark.svg
Comment (BenSouchet) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: I do not understand why the fact the SVG as multiple paths is an issue, lots of SVG have multiple paths, even for `plain` version of icon / logo.

I don't think reducing to one path will improve the file size a lot, and it will complicate things if changes need to be made on the SVG afterwards.

And for info, for these SVG files, even using the links you shared you won't be able to reduce the number of path since these paths cannot be merge (at least with my knowledge and understanding of the SVG specification).

Last point, **icomoon** when importing SVGs will convert them to font glyph so the number of paths doesn't matter.

Comment (Snailedlt) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: I'm not sure what the exact reasoning for joining paths is either, but I am trying to enforce it because it is a part of the [SVG Standards](https://github.com/devicons/devicon/wiki/SVG-Standards)
> The plain and line versions (with or without wordmark) need to stay as simple as possible. They must have only one color and the paths are united.

Maybe @Panquesito7 could tell us more about why that is, so we can update the wiki with more info (or remove it if it's not needed).


As for merging the paths, you're right. Doesn't work because the last path starts with `m` instead of `M`. It should still be possible in Inkscape though. I gave it a try and almost got there with this
matplotlib svg

almost identical, but the lines from the radar circle thingy are cutting through in some places

Here's both after eachother for comparison
matplotlib
matplotlib svg

Comment (BenSouchet) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: Yes the result of the "merge" give incorrect result and like I said the file size won't decrease, in fact the version you get is 3KB **(25%)** heavier than the one I created (with multiple paths)

Comment (BenSouchet) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: Having few / fewer paths is, lot of the time a good idea, **but** wanting to optimize more than what you can get with svg optimizers (like [SVG Viewer](https://www.svgviewer.dev/) or [SVGOMG](https://jakearchibald.github.io/svgomg/)) won't give you the expected result or will create unstable files

Comment (Snailedlt) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: @BenSouchet Alright, I think I got the paths merged so that it's identical now:
## New
image

## Old
matplotlib


## New & Old side by side

Comment (Snailedlt) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: > Having few / fewer paths is, lot of the time a good idea, **but** wanting to optimize more than what you can get with svg optimizers (like [SVG Viewer](https://www.svgviewer.dev/) or [SVGOMG](https://jakearchibald.github.io/svgomg/)) won't give you the expected result or will create unstable files

I don't have the knowledge to agree or disagree with you here, but your argument seems logical, so I'm just gonna wait and see what the other @devicons/maintainer s think about this one

Comment (BenSouchet) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: @Snailedlt Nice, yep it's seems they are identical
 
I compared file size of the one you just created and it's heavier (**16%**), so don't know also what is better but SVG I created for this project had rendering issue and like I said when **icomoon** convert the SVG the result is a glyph so I think the two version, mine and yours will give the exact same result.

So at the end, the SVG with only one path that is heavier and impossible to edit again (since every shape is merged) is maybe not the best one.

Yeah I'm also interested to see what the others maintainers think on that topic :)

Comment (Snailedlt) [reply icons/matplotlib/matplotlib-plain-wordmark.svg:1]: Yeah rom what I can tell multiple paths is better in this instance, but I'm hesitant to merge because of the SVG Standards like I mentioned earlier.

I guess we'll have to wait and see for the other maintainers

**Review by Snailedlt — APPROVED** [review]
Since we haven't heard anything from the other maintainers I think we're safe to merge this.

@devicons/maintainer
I'll ping one last time, and if there are no comments by tomorrow (same time as now), I'll go ahead and merge.

**Review by Panquesito7 — APPROVED** [review]
Looking great. Thanks! 🎉


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good&nbsp; | &nbsp;**Tags:** Efficient

**Pull request merged**
